### PR TITLE
auth and users controller

### PR DESCRIPTION
**Pull request by krystian-gawlik-fp**

### CODE REVIEW

File: src/controllers/auth.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      }
+
+      return jwt.sign(
+        {
+          id: user.id,
+          email: user.email,
+          role: user.role.toString()
+        },
+        process.env.JWT_SECRET,
+        { expiresIn: '1h' }
```
Comment (Miszczu2137) [inline src/controllers/auth.ts]: This value should comes from config.

File: src/controllers/users.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        email: r.email,
+        role: r.role
+      }
+    })
+  }
+
+  @Get(':id')
+  @Security('jwt', [Role.Admin, Role.User])
+  public async getUser(id: number): Promise<User> {
+    const result = await db().query(
```
Comment (Miszczu2137) [inline src/controllers/users.ts]: You shouldn't keep DB query into controller. Move this code into repository / dao and reiterate it for every controller.

File: src/middlewares/authentication.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  request: express.Request,
+  securityName: string,
+  scopes?: string[]
+): Promise<JwtAuthPayload> {
+  const token =
+    request.body.token ||
+    request.query.token ||
+    request.headers['authorization']
+
+  return new Promise((resolve, reject) => {
```
Comment (Miszczu2137) [inline src/middlewares/authentication.ts]: Please use async/await pattern instead of Promise.

File: src/controllers/auth.ts (hunk 1 above)
Comment (krystian-gawlik-fp) [reply src/controllers/auth.ts]: done

File: src/middlewares/authentication.ts (hunk 3 above)
Comment (krystian-gawlik-fp) [reply src/middlewares/authentication.ts]: done

File: src/controllers/users.ts (hunk 2 above)
Comment (krystian-gawlik-fp) [reply src/controllers/users.ts]: done

**Review by Miszczu2137 — APPROVED** [review]

**Pull request merged**